### PR TITLE
Teach the Syncer to drop blobs from target repositories

### PR DESCRIPTION
**Pull request by nulltoken**

### CODE REVIEW

**Comment by nulltoken** [comment]
@SimonCropp A first batch of change. Very crude.

Appveyor seems to cringe a bit

```
C:\projects\githubsync\src\GitHubSync\GitHubSync.csproj.metaproj : warning MSB4078: The project file "GitHubSync\GitHubSync.csproj" is not supported by MSBuild and cannot be built.
C:\projects\githubsync\src\Tests\Tests.csproj.metaproj : warning MSB4078: The project file "Tests\Tests.csproj" is not supported by MSBuild and cannot be built.
```

In order for me to ease the testing on my side, would you mind temporarily granting me with push bits on https://github.com/SimonCropp/GitHubSync.TestRepository ? :pray:

File: src/GitHubSync/MapperBase.cs [hunk 1]
```diff
@@ -79,11 +95,26 @@ IEnumerator IEnumerable.GetEnumerator()
...
+                    d.Add(orb, items);
+                }
+
+                items.Add(new Tuple<Parts, IParts>(destination, source));
+            }
+        }
+
+        foreach (var destination in toBeRemovedEntries)
+        {
+            {
```
Comment (SimonCropp) [inline src/GitHubSync/MapperBase.cs]: double braced?

**Comment by SimonCropp** [comment]
from my (naive) perspective this looks good

**Comment by nulltoken** [comment]
> from my (naive) perspective this looks good

@SimonCropp I don't think your perspective is naive in _any_ way :wink:

Besides the unit tests, I've ran other sets of integration tests ensuring that trees are (recursively) removed when the the last blob is dropped from them.

>I've still got some minor tweakings to perform but I believe most of the changes are there.

I think I'm done with them now.

Please run some tests on your side and report any issue you'll discover. I'll be happy to fix them.

**Comment by nulltoken** [comment]
@SimonCropp Thanks for granting me with push bits on the TestRepository. I've just updated the `target` branch with nested trees.

I then updated this PR to update `RepoSyncTests.SyncPr` integration test.

That led to the creation of https://github.com/SimonCropp/GitHubSync.TestRepository/pull/26 (peeling to the proposed commit that you can [browse](https://github.com/SimonCropp/GitHubSync.TestRepository/tree/dcf2aabc9a7a5572bea1e28781cf166ce6db22bf) into which `a` doesn't contain any sub folder anymore)

Looks like it's working.... Un-Wip-'d that PR which might be ready for final test and review

**Comment by SimonCropp** [comment]
going to merge this, do a beta release, and smoke test in a few places.

BTW awesome work as always

**Pull request merged**